src/lib.rs

```diff
@@ -0,0 +1,378 @@
+        .collect();
+    x
+}
+
+pub fn get_crate(db: &Connection, crate_name: &str) -> Result<Option<Crate>, rusqlite::Error> {
+    //Version id, Version num
+    let a = get_latest(db, crate_name.to_string())?;
+    let aa = a.last();
+    if aa.is_none() {
+        return Err(Error::InvalidQuery);
+    }
+    let version_id = &aa.unwrap().0;
+
+    let mut b = db.prepare_cached(
+        "SELECT crates.name, dependencies.req, dependencies.kind
+        FROM dependencies
+        LEFT JOIN crates
+            ON crates.id = dependencies.crate_id 
+        WHERE dependencies.version_id = ?",
+    )?;
+    let b_row = b.query_map([version_id], |r| {
+        Ok((
+            r.get_unwrap::<_, String>(0),
+            r.get_unwrap::<_, String>(1),
+            r.get_unwrap::<_, String>(2),
+        ))
+    })?;
+
+    let v3 = b_row
+        .filter(|f| f.is_ok())
+        .map(|f| f.unwrap())
+        .collect::<Vec<(String, String, String)>>();
+
+    let mut s = db.prepare_cached(
+        "SELECT id, name, description, downloads, homepage, repository, updated_at FROM crates WHERE name = ?",
+    )?;
+    let row = s.query_map([crate_name], |r| {
+        Ok((
+            r.get_unwrap::<_, String>(0),
+            r.get_unwrap::<_, String>(1),
+            r.get_unwrap::<_, String>(2),
+            r.get_unwrap::<_, String>(3),
+            r.get_unwrap::<_, String>(4),
+            r.get_unwrap::<_, String>(5),
+            r.get_unwrap::<_, String>(6),
+            get_keywords(db, r.get_unwrap::<_, String>(0)).unwrap_or_default(),
+        ))
+    })?;
+
+    let c = row
+        .filter(|f| f.is_ok())
+        .map(|f| f.unwrap())
+        .map(|f| Crate {
+            crateid: f.1,
+            disc: f.2,
+            downloads: (f.3).parse().unwrap_or_default(),
+            homepage_url: if f.4.is_empty() { None } else { Some(f.4) },
+            repo_url: if f.5.is_empty() { None } else { Some(f.5) },
+            last_update: f.6,
+            tags: f.7,
+            dependencies: v3.clone(),
+            ..Default::default()
+        });
+    Ok(c.into_iter().last())
+}
+
+#[derive(Default, Debug)]
+pub struct Crate {
+    pub crateid: String,
+    pub tags: Vec<String>,
+    pub versions: Vec<String>,
+    pub disc: String,
+    pub downloads: u32,
+    pub repo_url: Option<String>,
+    pub homepage_url: Option<String>,
+    pub last_update: String,
+    pub dependencies: Vec<(String, String, String)>,
+}
+
+#[cfg(feature = "capture-spantrace")]
+pub fn install_tracing() {
+    use tracing_error::ErrorLayer;
+    use tracing_subscriber::prelude::*;
+    use tracing_subscriber::{fmt, EnvFilter};
+
+    let fmt_layer = fmt::layer().with_target(false);
+    let filter_layer = EnvFilter::try_from_default_env()
+        .or_else(|_| EnvFilter::try_new("info"))
+        .unwrap();
+
+    tracing_subscriber::registry()
+        .with(filter_layer)
+        .with(fmt_layer)
+        .with(ErrorLayer::default())
+        .init();
+}
```

Comment:
Tracing setup should be handled by the binary, since this is now a library this serves no purpose.

Reply:
I've added it to the example